aiidalab_widgets_base/computational_resources.py

```diff
@@ -512,35 +515,35 @@ def _send_password(self, _=None):
     @traitlets.observe("ssh_connection_state")
     def _observe_ssh_connnection_state(self, _=None):
         """Observe the ssh connection state and act according to the changes."""
-        if self.ssh_connection_state == SshConnectionState.waiting_for_input:
+        if self.ssh_connection_state is SshConnectionState.waiting_for_input:
             return
-        if self.ssh_connection_state == SshConnectionState.success:
+        if self.ssh_connection_state is SshConnectionState.success:
             self.password_message = (
                 "The passwordless connection has been set up successfully."
             )
             return
-        if self.ssh_connection_state == SshConnectionState.keys_already_present:
+        if self.ssh_connection_state is SshConnectionState.keys_already_present:
             self.password_message = "The passwordless connection has already been setup. Nothing to be done."
             return
-        if self.ssh_connection_state == SshConnectionState.no_keys:
+        if self.ssh_connection_state is SshConnectionState.no_keys:
             self.password_message = (
                 " Failed\nLooks like the key pair is not present in ~/.ssh folder."
             )
             return
-        if self.ssh_connection_state == SshConnectionState.unknown_hostname:
+        if self.ssh_connection_state is SshConnectionState.unknown_hostname:
             self.password_message = "Failed\nUnknown hostname."
             return
-        if self.ssh_connection_state == SshConnectionState.connection_refused:
+        if self.ssh_connection_state is SshConnectionState.connection_refused:
             self.password_message = "Failed\nConnection refused."
             return
-        if self.ssh_connection_state == SshConnectionState.end_of_file:
+        if self.ssh_connection_state is SshConnectionState.end_of_file:
             self.password_message = (
                 "Didn't manage to connect. Please check your username/password."
             )
             return
-        if self.ssh_connection_state == SshConnectionState.enter_password:
+        if self.ssh_connection_state is SshConnectionState.enter_password:
             self._handle_ssh_password()
-        elif self.ssh_connection_state == SshConnectionState.do_you_want_to_continue:
+        elif self.ssh_connection_state is SshConnectionState.do_you_want_to_continue:
             self._ssh_connection_process.sendline("yes")
 
         self.ssh_connection_state = SshConnectionState.waiting_for_input
```

Comment:
Hi @csadorf, I think this `waiting_for_input` is needed to bookkeep a 'holding' state when setting password. 
If we don't have this modify of `ssh_connection_state` when I click `Setup ssh` twice the setting process will stuck, since the `ssh_connection_state` end up at `nter_password` state.

Since for my understanding `waiting_for_input` is waiting for password input, I move it to `ssh_handle`, and seems works fine.